tests/unit/conftest.py

```diff
@@ -16,19 +16,21 @@
 
 import mock
 import pytest
+from juju.client._definitions import ApplicationStatus, UnitStatus
 
 from cou.apps.app import OpenStackApplication
+from cou.apps.auxiliary import OpenStackAuxiliaryApplication
 from cou.apps.subordinate import OpenStackSubordinateApplication
 
 
 @pytest.fixture
 def status():
-    mock_keystone_ussuri = mock.MagicMock()
+    mock_keystone_ussuri = mock.MagicMock(spec=ApplicationStatus)
```

Comment:
I didn't know about this feature. Very cool @rgildein

I think we should use `spec_set` instead of `spec` based in this [discussion](https://stackoverflow.com/questions/25323361/what-is-spec-and-spec-set)

`spec` still allows us to set attribute that is not specified, while with spec_set, it is not allowed to set unspecified attribute.

Reply:
done